actionpack/test/dispatch/cookies_test.rb

```diff
@@ -1274,30 +1274,120 @@ def test_cookies_are_not_cleared
   end
 
   def test_signed_cookie_with_expires_set_relatively
+    switch_on_cookies_with_metadata
+
     cookies.signed[:user_name] = { value: "assain", expires: 2.hours }
 
     travel 1.hour
     assert_equal "assain", cookies.signed[:user_name]
 
     travel 2.hours
     assert_nil cookies.signed[:user_name]
+
+  ensure
+    switch_off_cookies_with_metadata
```

Comment:
Having these ensures spread throughout is a bad practice.

Switch to:

```ruby
with_cookie_metadata_enabled do
  # …
end
```

Remember to capture the old value in a local variable too and reassign it back.